pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -622,21 +640,34 @@ func (exp *FlowExporter) findFlowType(conn flowexporter.Connection) uint8 {
 	}
 
 	if exp.nodeRouteController == nil {
-		klog.V(4).InfoS("Can't find flowType without nodeRouteController")
-		return 0
+		if klog.V(5).Enabled() {
+			klog.InfoS("Can't find flowType without nodeRouteController")
+		}
+		return flowTypeUnknown
+	}
+	srcIsPod, srcIsGw := exp.nodeRouteController.LookupIPInPodSubnets(conn.FlowKey.SourceAddress)
+	dstIsPod, dstIsGw := exp.nodeRouteController.LookupIPInPodSubnets(conn.FlowKey.DestinationAddress)
+	if srcIsGw || dstIsGw {
+		// This matches what we do in filterAntreaConns but is more general as we consider
+		// remote gateways as well.
+		if klog.V(5).Enabled() {
+			klog.InfoS("Flows where the source or destination IP is a gateway IP will not be exported")
+		}
+		return flowTypeUnsupported
 	}
-	if exp.nodeRouteController.IPInPodSubnets(conn.FlowKey.SourceAddress.AsSlice()) {
-		if conn.Mark&openflow.ServiceCTMark.GetRange().ToNXRange().ToUint32Mask() == openflow.ServiceCTMark.GetValue() || exp.nodeRouteController.IPInPodSubnets(conn.FlowKey.DestinationAddress.AsSlice()) {
-			if conn.SourcePodName == "" || conn.DestinationPodName == "" {
-				return ipfixregistry.FlowTypeInterNode
-			}
-			return ipfixregistry.FlowTypeIntraNode
+	if !srcIsPod {
+		if klog.V(5).Enabled() {
```

Comment:
So we haven't supported external to pod flow, right?

Reply:
No, we have never supported this case, and we still don't.